definition_validator/variance.cc

```diff
@@ -0,0 +1,184 @@
+
+#include "common/typecase.h"
+#include "core/Symbols.h"
+#include "core/core.h"
+#include "core/errors/resolver.h"
+
+#include "variance.h"
+
+using namespace std;
+
+namespace sorbet::definition_validator::variance {
+
+class VarianceValidator {
+private:
+    const core::Loc loc;
+
+    VarianceValidator(const core::Loc loc) : loc(loc) {}
+
+    core::Variance invertVariance(const core::Variance variance) {
+        switch (variance) {
+            case core::Variance::CoVariant:
+                return core::Variance::ContraVariant;
+            case core::Variance::Invariant:
+                return core::Variance::Invariant;
+            case core::Variance::ContraVariant:
+                return core::Variance::CoVariant;
+        }
+    }
+
+    string showVariance(const core::Variance variance) {
+        switch (variance) {
+            case core::Variance::CoVariant:
+                return ":out";
+            case core::Variance::Invariant:
+                return ":invariant";
+            case core::Variance::ContraVariant:
+                return ":in";
+        }
+    }
+
+    bool compatibleVariance(const core::Variance left, const core::Variance right) {
```

Comment:
lets make `left` into `expected` and give it a type called `polarity` and lets make `right` into `SymbolRef`?
This will make for a method that's easier to read

Reply:
also, let's call method with a verb: `hasCompatibleVariance`